src/mpyl/cli/artifacts.py

```diff
@@ -0,0 +1,105 @@
+"""Commands to manage remotely cached artifacts"""
+
+from pathlib import Path
+from tempfile import TemporaryDirectory
+import logging
+from pyaml_env import parse_config
+import click
+
+from ..artifacts.build_artifacts import BuildArtifacts
+from ..cli import CliContext, create_console_logger, CONFIG_PATH_HELP
+from ..constants import DEFAULT_CONFIG_FILE_NAME, DEFAULT_RUN_PROPERTIES_FILE_NAME
+from ..steps.models import RunProperties
+from ..utilities.repo import Repository, RepoConfig
+
+
+@click.group("artifacts")
```

Comment:
I would expect `push` and `pop` to be filed under `build` intuitively.
The workflow would be
```
mpyl build pop
mpyl build run
mpyl build push
```
WDYT?

Reply:
Well this is how we refined it a few weeks ago during our call, we decided to implement it in 2 ways:
- A toggle to the build commands
- Its own cli commands

I did look into creating an 'artifacts' subgroup in `build` but that didn't seem possible.

Reply:
On second thought I prefer the way I suggested. Don't you?

Reply:
Eh it's just semantics really, programmatically I prefer `mpyl artifacts pull/push` since `build` is already big enough by itself. It's also more explicit to what you're actually doing compared to `build pop`. But it's not a big deal for me, you can decide with @koeves next week since he'll take over the pr while I'm on vacation.